Lib/test/support/os_helper.py

```diff
@@ -716,3 +717,37 @@ def __exit__(self, *ignore_exc):
             else:
                 self._environ[k] = v
         os.environ = self._environ
+
+
+try:
+    import ctypes
+    kernel32 = ctypes.WinDLL('kernel32', use_last_error=True)
+except (ImportError, AttributeError):
+    def subst_drive(path):
+        raise NotImplementedError('ctypes or kernel32 is not available')
+else:
+    ERROR_FILE_NOT_FOUND = 2
+    DDD_REMOVE_DEFINITION = 2
+    DDD_EXACT_MATCH_ON_REMOVE = 4
+    DDD_NO_BROADCAST_SYSTEM = 8
+
+    @contextlib.contextmanager
+    def subst_drive(path):
+        """Temporarily yield a substitute drive for a given path."""
+        for c in reversed(string.ascii_uppercase):
+            drive = f'{c}:'
+            if (not kernel32.QueryDosDeviceW(drive, None, 0) and
+                    ctypes.get_last_error() == ERROR_FILE_NOT_FOUND):
+                break
+        else:
+            raise FileExistsError('no available logical drive')
+        if not kernel32.DefineDosDeviceW(
+                DDD_NO_BROADCAST_SYSTEM, drive, path):
+            raise ctypes.WinError(ctypes.get_last_error())
+        try:
+            yield drive
+        finally:
+            if not kernel32.DefineDosDeviceW(
+                    DDD_REMOVE_DEFINITION | DDD_EXACT_MATCH_ON_REMOVE,
```

Comment:
```suggestion
                    DDD_REMOVE_DEFINITION | DDD_EXACT_MATCH_ON_REMOVE | DDD_NO_BROADCAST_SYSTEM,
```

Reply:
I left out `DDD_NO_BROADCAST_SYSTEM` in case an app has manually refreshed its list of drives via `GetLogicalDrives()` or `GetLogicalDriveStringsW()`, such as by pressing F5 in an Explorer window. For example, broadcasting the `WM_DEVICECHANGE` message lets Explorer automatically remove the drive icon from "This PC". I'm not really concerned with what applications might do when notified that a drive has been removed. This is just a courtesy for individuals running the test suite.

Reply:
Makes perfect sense, thank you.